ariac_plugins/src/agv_tray_plugin.cpp

```diff
@@ -164,6 +167,13 @@ void AGVTrayPlugin::OnUpdate()
     }
   }
 
+  // Unlock tray if agv is at the warehouse
+  if (abs(impl_->agv_joint_->Position(0) - impl_->warehouse_location_) < 0.3 && impl_->tray_attached_) {
```

Comment:
Why don't we use the topic which reports the AGV location? If we change the environment in the future, we will need to hardcode new values.

Reply:
If the competitor submits the order immediately there can be a race condition where the tray was not properly unlocked and teleporting it breaks the AGV.